tests/WatchTests.cs

```diff
@@ -326,13 +316,11 @@ private class DummyHandler : DelegatingHandler
         [Fact]
         public void TestWatchWithHandlers()
         {
+            using (CountdownEvent eventsReceived = new CountdownEvent(1))
```

Comment:
suggest remove using due eventsReceived is referenced by closure.

Reply:
Using means disposal is in reverse order of declaration so I think this should be safe.

Reply:
Agree, I believe this shoudl be safe as the watched will be deleted before the eventsReceived object.